app/src/backend/formulas.ts

```diff
@@ -115,6 +117,80 @@ const DEFAULT_INCOME_GROUP_FRACTIONS: Record<string, number> = {
   "income-group-type-urban-low-income": 0.27,
 };
 
+// check if an extra field has a unit.
+// if yes. take the selected unit, convert it to the default unit and return the value
+
+// take in the activityData and the methodology
+//
+function convertDataToDefaultUnit(
+  activityValue: ActivityValue,
+  methodologyId: string,
+  referenceNumber: string,
+): Record<string, any> {
+  const methododology = findMethodology(methodologyId, referenceNumber);
+  if (!methododology) {
+    throw new createHttpError.NotFound(
+      `Could not find methodology for reference number ${referenceNumber}`,
+    );
+  }
+
+  let activity = methododology.activities?.[0];
+
+  // if methodologyId is !direct measure, and number of activities === 1 use the Oth activity
+  if ((methododology.activities?.length as number) > 1) {
+    let selectedActivityOption =
+      activityValue.metadata?.[
+        methododology.activitySelectionField?.id as string
+      ];
+
+    const foundIndex =
+      methododology.activities?.findIndex(
+        (ac) => ac.activitySelectedOption === selectedActivityOption,
+      ) ?? 0;
+
+    const selectedActivityIndex = foundIndex >= 0 ? foundIndex : 0;
+
+    activity = methododology.activities?.[selectedActivityIndex];
+  }
+  // deal with activity title value
+
+  let data: Record<string, any> = { ...activityValue.activityData };
+  // check if it has a default unit property
+  if (activity?.["default-units"]) {
+    const val = data[activity?.["activity-title"] as string];
+    const fuelTypeKey = Object.keys(data).find((key) =>
+      key.includes("fuel-type"),
+    );
+    const fuelType = data[fuelTypeKey as string];
+    data[activity?.["activity-title"] as string] = new Decimal(
+      UnitConversionService.convertUnits(
+        val,
+        data[`${activity?.["activity-title"]}-unit`],
```

Comment:
Could you maybe extra this key into a variable beforehand instead of doing string interpolation in the object access in the function call?